examples/chatgpt/gpt_actions_library/gpt_action_snowflake_middleware.ipynb

```diff
@@ -0,0 +1,686 @@
+    "  title: Snowflake GPT API\n",
+    "  description: API to execute SQL queries on Snowflake and get the results as a CSV file URL.\n",
+    "  version: 1.0.0\n",
+    "servers:\n",
+    "  - url: https://<server-name>.azurewebsites.net\n",
+    "    description: Azure Function App server running Snowflake integration application\n",
+    "paths:\n",
+    "  /api/<function_name>?code=<code>:\n",
+    "    post:\n",
+    "      operationId: executeSQL\n",
+    "      summary: Executes a SQL query on Snowflake and returns the result file URL as a CSV.\n",
+    "      requestBody:\n",
+    "        required: true\n",
+    "        content:\n",
+    "          application/json:\n",
+    "            schema:\n",
+    "              type: object\n",
+    "              properties:\n",
+    "                sql_query:\n",
+    "                  type: string\n",
+    "                  description: The SQL query to be executed on Snowflake.\n",
+    "              required:\n",
+    "                - sql_query\n",
+    "      responses:\n",
+    "        '200':\n",
+    "          description: Successfully executed the query.\n",
+    "          content:\n",
+    "            application/json:\n",
+    "              schema:\n",
+    "                type: object\n",
+    "                properties:\n",
+    "                  openaiFileResponse:\n",
+    "                    type: array\n",
+    "                    items:\n",
+    "                      type: string\n",
+    "                      format: uri\n",
+    "                    description: Array of URLs pointing to the result files.\n",
+    "        '401':\n",
+    "          description: Unauthorized. Missing or invalid authentication token.\n",
+    "        '400':\n",
+    "          description: Bad Request. The request was invalid or cannot be otherwise served.\n",
+    "        '500':\n",
+    "          description: Internal Server Error. An error occurred on the server.\n",
+    "components:\n",
+    "  schemas: {} "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### FAQ & Troubleshooting"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "* Files returned to ChatGPT are limited in size to 10MB. Your request may fail if the file returned is larger than that. Ensure to include LIMITs on your SQL commands if you find you are running into these limitations.\n",
+    "* _Why is the Azure Function App requred in the first place?_ ChatGPT’s Data Analysis feature (aka Code Interpreter) depends on a secure Python environment that is separate from the model’s context window. Data passed to Data Analysis must be done so by uploading a file today. GPT actions returning data must then return that data as a CSV or other data file type. In order to return a file via GPT action, the response must be wrapped in an `openaiFileResponse` object. This requires custom code to properly format the response.\n",
```

Comment:
Curious on why the Drive integration can return files directly if we have a specific format expected. We don't need a middleware for that one I believe?

Reply:
It doesn't return files. It returns file contents. You can't pass a file directly from Drive into Code Interpreter without middleware